PathSum2.java

```diff
@@ -0,0 +1,59 @@
+// Time Complexity : O(n)
```

Comment:
Kindly check time complexity. Also add solution using backtracking as well as that is the optimized solution

Reply:
done